### DISCUSSION TITLE
Domains are deprecated, and so should RequestContext.

### DISCUSSION THREAD

**Discussion by iyobo** [mikro-orm/mikro-orm#1590] [General]
What's the best way to fork the orm in such a way that I'm not passing an entityManager all over the place e.g as parameters to various service functions, etc.

I see the `RequestContext` being mentioned, but I understand that uses NodeJS Domains, which are deprecated.
https://nodejs.org/api/domain.html

This seems to leave us with orm.fork().

So would it mean each service function should fork it's own context?

```

api.get('path',(...)=>{
   const em = orm.fork()
   ...
   await someServicefunction()
   ...
   await em.flush();
})

async function someServicefunction(){
   const em = orm.fork();
   ...
   await em.flush();
}

```

That seems like a lot of forking. Also What is the memory/cpu profile of a fork?

**Comment by Langstra** [comment]
Would asyncLocalStorage be an option? It is documented here: https://mikro-orm.io/docs/async-local-storage/

**Reply by B4nan** [reply]
Btw ALS will replace current domain api solution in v5. 

If you dont like this approach, fork manually, but it might be tedious, especially when you want to use DI containers. 

With domain api there is some performance cost, but ALS was much faster. I dont have any numbers, but you should be able to measure and compare it yourself, should be trivial. 

Cost of having new context per request is small, most of the classes are shared across forks, its mainly the EM and UoW, no heavy resources. They will get garbage collected after the request is resolved.

**Comment by iyobo** [comment]
@B4nan Glad to see `AsyncLocalStorage` exists pending removal of Domain.

But I don't really see a whole lot of documentation behind it.
E.g, how does it get used in a route? Or a service function as shown in OP?
